### PR TITLE
Revamp the introduction page

### PR DESCRIPTION
**Pull request by Zethson**
This isn't API feedback, just intro page feedback.

I am still not sure how much value the `introduction` page actually has since it mostly shows off random pieces of code in the `Quickstart` section without any context, justification or story.

Goals that I want to achieve **here**:

1. Add at least one sentence of context at the very top to justify why the hell lamindb exists in the first place. Inspired by latch.
2. Reduce the `Design` section which is IMO too long.
3. Simplify the language. chatGPT didn't know what a mesh of databases is so why would random people?

### CODE REVIEW

**Comment by falexwolf** [comment]
> chatGPT didn't know what a mesh of databases is so why would random people?

We're not targeting random people here, but data architects in biotech. Among those, it's a well known fact that the industry moved away from centralized to mesh/distributed systems. The word appears in the 5th bullet and is the only one that tries to target this highly technical audience. The fact that lamindb is a distributed/mesh system is very unique. Previously we used the word "distributed", but because "mesh" came up a few times in conversations, I changed it.

Quilt has it even on the repo-level description (https://github.com/quiltdata/quilt):
image

And Quilt is so far the most popular solution for data lakes in the whole industry.

So, why can't we have this word (or "distributed") in bullet 5?

File: README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@
 - Manage storage & databases with a unified Python API ("lakehouse").
 - Track data lineage across notebooks & pipelines.
 - Integrate registries for experimental metadata & in-house ontologies.
-- Validate, standardize & annotate.
-- Collaborate across a mesh of distributed LaminDB instances.
```
Comment (falexwolf) [inline README.md:11]: See my note below. "mesh" is a well-known keyword in the field that triggers a certain connotation for the most important reader audience.

But given this is a simplification and maintains "distributed", I think it's OK to risk not triggering the connotation. People who understand mesh should also understand that it's implied by "distributed".

File: README.md [hunk 2]
```diff
@@ -7,8 +7,8 @@
...
+- Validate, standardize & annotate metadata.
```
Comment (falexwolf) [inline README.md:10]: We don't annotate metadata, we annotate data.

I'd not add "metadata", here. We had "data" here before it felt quite repetitive (what else would you validate, standardize & annotate).

File: README.md (hunk 1 above)
Comment (falexwolf) [reply README.md:11]: Thanks for spotting this!

File: README.md (hunk 2 above)
Comment (Zethson) [reply README.md:10]: Whops, good catch. Yes of course

**Comment by Zethson** [comment]
> So, why can't we have this word (or "distributed") in bullet 5?

We can and totally should use `distributed`, but `mesh` in my opinion is not a common enough term to describe `distributed` in this context.

I'll happily argue that Quilt is doing themselves a disservice by using `mesh`.

**Comment by Zethson** [comment]
Besides the details, is this PR generally going into a direction that you agree with or would you rather not have it?

**Comment by falexwolf** [comment]
> Add at least one sentence of context at the very top to justify why the hell lamindb exists in the first place. Inspired by latch.

image

I don't like adding this or probably any sentence.

We have a whole landing page at lamin.ai that explains why Lamin exists.

image
image
image
image
image
image

These are the 6 reasons what we should iterate on (Maybe there should be 3,4,5, or 7 reasons? Are these really the perfect words that nail the problems we solve? ...). There is a ton of improvement potential on the landing page.

Everbody knows that data in bio is mismanaged, and I know enough people who get annoyed when this gets pointed out in generic ways:

image

I think this is a hella long paragraph that would be suited for a paper about lamindb, but not for somebody who just came from the landing page at lamin.ai to lamin.ai/docs.

**Comment by falexwolf** [comment]
> We can and totally should use distributed, but mesh in my opinion is not a common enough term to describe distributed in this context.
> I'll happily argue that Quilt is doing themselves a disservice by using mesh.

Happy to agree on this! Mesh is weaker than distributed, but I agree this nuance is for deep down in the docs.

**Comment by falexwolf** [comment]
I now made it until down here:

image

I liked everything including moving the "Assumptions" further up!

**Comment by falexwolf** [comment]
I wouldn't say this:

image

We're not open-sourcing almost all our work because at least a comparable amount of work is done for the hub, and that's closed source.

I'd also say that "pride" rings an ideological bell, which I'd like to avoid. I think it's perfectly OK to simply use plain language (as it was) and then people can decide what they think about it. We're open-sourcing the core data access & data modeling parts to
1. put users in full control of their data
2. help defragmenting data around the world

These are very pragmatic reasons.

**Comment by Zethson** [comment]
> > Add at least one sentence of context at the very top to justify why the hell lamindb exists in the first place. Inspired by latch.
> 
> image
> 
> I don't like adding this or probably any sentence.
> 
> We have a whole landing page at lamin.ai that explains why Lamin exists.
> image image image image image image
> 
> These are the 6 reasons what we should iterate on (Maybe there should be 3,4,5, or 7 reasons? Are these really the perfect words that nail the problems we solve? ...). There is a ton of improvement potential on the landing page.
> 
> Everbody knows that data in bio is mismanaged, and I know enough people who get annoyed when this gets pointed out in generic ways. I fear the following risks annoying readers, too:
> image
> 
> I think this is a hella long paragraph that would be suited for a paper about lamindb, but not for somebody who just came from the landing page at lamin.ai to lamin.ai/docs.

All right, fair.
I took this as an opportunity to check how a few other tools are doing this:

MLFlow:

> MLflow is an open-source platform, purpose-built to assist machine learning practitioners and teams in handling the complexities of the machine learning process. MLflow focuses on the full lifecycle for machine learning projects, ensuring that each phase is manageable, traceable, and reproducible.

W & B:

> Weights & Biases is the machine learning platform for developers to build better models faster. Use W&B's lightweight, interoperable tools to quickly track experiments, version and iterate on datasets, evaluate model performance, reproduce models, visualize results and spot regressions, and share findings with colleagues.

Latch bio:

> It takes months to build infrastructure with the compute, storage, and user-friendly interface necessary to run bioinformatics pipelines at scale.

The Latch SDK is an open-source toolchain to define serverless bioinformatics workflows with plain python and deploy associated no-code interfaces using single command. [...]

Neptune:

> Neptune is the MLOps stack component for experiment tracking. It offers a single place to log, compare, store, and collaborate on experiments and models.

DVC:

> DVC is a tool for data science that takes advantage of existing software engineering toolset. It helps machine learning teams manage large datasets, make projects reproducible, and collaborate better.

I could continue. I think one trend that I notice is that they justify in their 1-2 intro sentences why they exist. We don't:

> LaminDB is an open-source data framework for biology.

Therefore, I propose that we slightly adapt our "LaminDB is an open-source data framework for biology." sentence to include a justification and remove the other fluff that I added.

**Comment by falexwolf** [comment]
I agree that this here doesn't have to be on the intro page, but it gotta move somewhere:

image

Can you move it somewhere so that the information doesn't get lost?

**Comment by falexwolf** [comment]
I made it to the end! I like all the changes except the long sentence at the very top. :D 

Thanks, Lukas! 🙏

**Comment by Zethson** [comment]
Thanks!
Curious about your thoughts on https://github.com/laminlabs/lamindb/pull/1514#issuecomment-2017918774. I'll make the other suggested/requested changes of course.

**Comment by Zethson** [comment]
> I agree that this here doesn't have to be on the intro page, but it gotta move somewhere:
> image
> 
> Can you move it somewhere so that the information doesn't get lost?

It's still there though under `repositories`. I wanted to avoid that people find `guides and tutorials` and then click on the links there and think this is how they're supposed to look at them.

**Comment by falexwolf** [comment]
Re the "intro sentence".

I randomly picked "W&B". After coming from their landing page, I end up on their github readme and find

> Use W&B to build better models faster. Track and visualize all the pieces of your machine learning pipeline, from datasets to production machine learning models.

Then I go to the docs, and I find the beast of a sentence that you reference:

> Weights & Biases is the machine learning platform for developers to build better models faster. Use W&B's lightweight, interoperable tools to quickly track experiments, version and iterate on datasets, evaluate model performance, reproduce models, visualize results and spot regressions, and share findings with colleagues.

Compare this with our GitHub Readme that also appears on the docs landing page:

> LaminDB is a data framework for biology:
> - Manage storage & databases with a unified Python API ("lakehouse").
> - Track data lineage across notebooks & pipelines.
> - Integrate registries for experimental metadata & in-house ontologies.
> - Validate, standardize & annotate.
> - Collaborate across distributed LaminDB instances.

I'd say you get the simpler and more consistent developer-oriented messaging on our GitHub/docs compared to W&B.

If we don't like bullets, we can try putting the bullets into a sentence that'd be similarly long as the sentence of W&B or scanpy. I prefer the 5 bullets instead of an equivalent long sentence. There are a few tools out there that list bullets as the first thing rather than a long sentence.

Modal:

image

Chroma:

image

Etc.

> I could continue. I think one trend that I notice is that they justify in their 1-2 intro sentences why they exist.

So, I think you're somehow not acknowledging that we have 5 concise bullets that explain why LaminDB exists, and you seem to prefer to put the content of these bullets into a single sentence. And we can debate that! But adding a mammoth sentence in addition to the 5 bullets doesn't make sense to me.

image

> The Latch SDK is an open-source toolchain to define serverless bioinformatics workflows with plain python and deploy associated no-code interfaces using single command.

> Neptune is the MLOps stack component for experiment tracking. It offers a single place to log, compare, store, and collaborate on experiments and models.

Re content: Just like Latch's & Neptune's & W&B's and a few more of your sentences, our 5 bullets describe what LaminDB does. There is no additional attempt to describe a "why". The other examples you mention try to mix in a bit of a problem statement. I'd stay away from this in our case, because our product is even more complicated than literally all of the examples your giving in the sense that LaminDB's scope is broader than each of the tools that you're listing. This makes messaging so hard because we can't just write down a single sentence that everybody understands. (It was really simple to come up with the sentence of Scanpy.)

**Comment by falexwolf** [comment]
> It's still there though under repositories. I wanted to avoid that people find guides and tutorials and then click on the links there and think this is how they're supposed to look at them.

OK! All good then! It's not so easy to follow the changes in the design section.

**Comment by Zethson** [comment]
Discussion continued here: https://laminlabs.slack.com/archives/C04FPE8V01W/p1711372317778849

**Comment by falexwolf** [comment]
One thing I only spotted now, where I'm also not clear on the value add, is this:

image

Isn't it clear that we'll show off "high level features"? I mean, compare this to say, Chroma. Can we just get going and show what you can do with LaminDB rather than announcing that we'll show what we're going to do?

I think that in this instance you made a simple sentence that has the same information in it more complicated. Am I off?

**Comment by Zethson** [comment]
Sorry, I don't get my mails properly at Helmholtz (it blocks my email provider). 

Since this introduction page in general doesn't really have any story and reads a bit "list-like" instead of "motivated-by-like", I think that specifying that this is just showing off the high level features and the rest can be found somewhere else, can be useful. 

However, I don't hold this change dear to my heart at all so if you disagree feel free to revert this. No problem!

**Pull request merged**